server/src/users/users.controller.spec.ts

```diff
@@ -1,23 +1,46 @@
 import { Test, TestingModule } from '@nestjs/testing';
-
+import { userCreateDtoStub } from '../../test/stubs';
 import { UsersController } from './users.controller';
 import { UsersService } from './users.service';
 
+jest.mock('./users.service');
+
 describe('UsersController', () => {
   let controller: UsersController;
+  let usersService: UsersService;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       controllers: [UsersController],
-      //TODO I put a empty mock for UserService. For real testing we'll need
-      // to create a mock since we are using TypeORM
-      providers: [{ provide: UsersService, useValue: {} }],
+      providers: [UsersService],
     }).compile();
 
     controller = module.get<UsersController>(UsersController);
+    usersService = module.get<UsersService>(UsersService);
+    jest.clearAllMocks();
```

Comment:
> why mock initially on line 6 if we clear here?

See: https://stackoverflow.com/a/66361187/13862254 
`jest.clearAllMocks()` resets the number of times a spy has been called. I't needed to keep them from polluting other subsequent tests.

Reply:
> in the docs i don't see them mocking at all actually https://docs.nestjs.com/fundamentals/testing#unit-testing i wonder if this `Test.createTestingModule` creates mocks under the hood for us

So, ` Test.createTestingModule` is just a utility for bootstrapping a test module, and injects dependencies into the class under test. In the [nestjs docs](https://docs.nestjs.com/fundamentals/testing#testing-utilities) they do mock - see this line when you follow the link:

**This is basically mocking an implementation ad hoc. The proposed way in this PR is to create a stub class instead of ad hoc mocks and spies**
`jest.spyOn(catsService, 'findAll').mockImplementation(() => result);`

Reply:
Marking as resolved, because I'm shifting into integration tests rather than unit